editoast/src/schema/switch_type.rs

```diff
@@ -53,6 +63,199 @@ impl Cache for SwitchType {
+        "point_switch" => {
+            groups.insert(
+                "A_B1".into(),
+                vec![SwitchPortConnection {
+                    src: "A".into(),
+                    dst: "B1".into(),
+                }],
+            );
+            groups.insert(
+                "A_B2".into(),
+                vec![SwitchPortConnection {
+                    src: "A".into(),
+                    dst: "B2".into(),
+                }],
+            );
+        }
+        "crossing" => {
+            groups.insert(
+                "STATIC".into(),
+                vec![
+                    SwitchPortConnection {
+                        src: "A1".into(),
+                        dst: "B1".into(),
+                    },
+                    SwitchPortConnection {
+                        src: "B2".into(),
+                        dst: "A2".into(),
+                    },
+                ],
+            );
+        }
+        "link" => {
+            groups.insert(
+                "STATIC".into(),
+                vec![SwitchPortConnection {
+                    src: "A".into(),
+                    dst: "B".into(),
+                }],
+            );
+        }
+        "double_slip_switch" => {
+            groups.insert(
+                "A1_B1".into(),
+                vec![SwitchPortConnection {
+                    src: "A1".into(),
+                    dst: "B1".into(),
+                }],
+            );
+            groups.insert(
+                "A1_B2".into(),
+                vec![SwitchPortConnection {
+                    src: "A1".into(),
+                    dst: "B2".into(),
+                }],
+            );
+            groups.insert(
+                "A2_B1".into(),
+                vec![SwitchPortConnection {
+                    src: "A2".into(),
+                    dst: "B1".into(),
+                }],
+            );
+            groups.insert(
+                "A2_B2".into(),
+                vec![SwitchPortConnection {
+                    src: "A2".into(),
+                    dst: "B2".into(),
+                }],
+            );
+        }
+        "single_slip_switch" => {
+            groups.insert(
+                "STATIC".into(),
+                vec![
+                    SwitchPortConnection {
+                        src: "A1".into(),
+                        dst: "B1".into(),
+                    },
+                    SwitchPortConnection {
+                        src: "A2".into(),
+                        dst: "B2".into(),
+                    },
+                ],
+            );
+            groups.insert(
+                "A1_B2".into(),
+                vec![SwitchPortConnection {
+                    src: "A1".into(),
+                    dst: "B2".into(),
+                }],
+            );
+        }
+        _ => panic!("Incorrect switch_type"),
+    };
+    groups
+}
+
+trait BuiltinType {
+    const TYPE: &'static str;
+    const PORTS: &'static [&'static str];
```

Comment:
You could add something like that to define the connections:

```rust
const CONNECTIONS: [[&'static str; 2]];
```

and iter on that to obtain what `builtin_node_groups` provide statically